docs/diagnostics/PX1096.md

```diff
@@ -0,0 +1,107 @@
+# PX1096
+This document describes the PX1096 diagnostic.
```

Comment:
@EAndrosova please check the provided documentation for the new diagnostic PX1096

Reply:
Please replace the diagnostic description with the following: 

# PX1096
This document describes the PX1096 diagnostic.

## Summary

| Code   | Short Description                                                                     | Type  | Code Fix    | 
| ------ | ------------------------------------------------------------------------------------- | ----- | ----------- | 
| PX1096 |Signature of a method with the `PXOverride` attribute must match the overriden method. | Error | Unavailable | 

## Diagnostic Description
Signature of a method with the `PXOverride` attribute must match the overriden method.
The signatures of the overriden and overriding methods must comply with one of the following rules:
* Signatures of both methods are identical.
* Signatures of both methods are identical, except the overriding method has one additional parameter, which is a delegate that points to the base method. In this scenario, the delegate signature must also match the overriden method.

## Further Constraints
* The overriding method cannot be static
* The overriding method cannot be generic
* The base method must be a member of graph extension, that is a class that derives from `PXGraphExtension`
* The base method must be virtual (have either the virtual or the override signature)
* The base method must have one of the following accessibility levels: public, protected or protected internal
* The names of the derived method and the base method must match

## Example of Correct Code

```C#
public class MyGraph : PXGraph<MyGraph>
public class BaseGraphExtension : PXGraphExtension<MyGraph>
{
	public virtual int Add(int x, string y)
	{
		return x + Convert.ToInt32(y);
	}
}
public class DerivedGraphExtension : PXGraphExtension<BaseGraphExtension, MyGraph>
{
	[PXOverride]
	public virtual int Add(int x, string y)
	{
		return x + Convert.ToInt32(y) * 2;
	}
}
```

## Example of Correct Code with Delegate

```C#
public class MyGraph : PXGraph<MyGraph>
public class BaseGraphExtension : PXGraphExtension<MyGraph>
{
	public virtual int Add(int x, string y)
	{
		return x + Convert.ToInt32(y);
	}
}
public class DerivedGraphExtension : PXGraphExtension<BaseGraphExtension, MyGraph>
{
	[PXOverride]
	public virtual int Add(int x, string y, Func<int, string, int> del)
	{
		if (x < 10)
		{
			return x + Convert.ToInt32(y) * 2;
		}
		
		return del(x, y);
	}
}
```

## Example of Incorrect Code
In the following example, the delegate signature does not correspond to the signature of the base method.
```C#
public class MyGraph : PXGraph<MyGraph>
public class BaseGraphExtension : PXGraphExtension<MyGraph>
{
	public virtual int Add(int x, string y, bool z)
	{
		return x + Convert.ToInt32(y);
	}
}
public class DerivedGraphExtension : PXGraphExtension<BaseGraphExtension, MyGraph>
{
	[PXOverride]
	public virtual int Add(int x, string y, bool z, Func<int, string, int> del)
	{
		if (x < 10 && z)
		{
			return x + Convert.ToInt32(y) * 2;
		}
		
		return del(x, y);
	}
}
```

## Related Articles
[To override a Virtual Method](https://help.acumatica.com/(W(16))/Help?ScreenId=ShowWiki&pageid=6fa2a444-17b4-42f9-9e6a-64e85167626a)
[PXOverrideAttribute](https://help.acumatica.com/Help?ScreenId=ShowWiki&pageid=4bd32f02-a673-3a01-1748-33a54f3cd035)

Reply:
Updated. Thank you.